pkgs/build-support/cc-wrapper/ld-wrapper.sh

```diff
@@ -135,10 +141,10 @@ if [ "$NIX_DONT_SET_RPATH" != 1 ]; then
     # so, add the directory to the rpath.
     # It's important to add the rpath in the order of -L..., so
     # the link time chosen objects will be those of runtime linking.
-    for i in ${libDirs[@]}; do
-        for j in ${libs[@]}; do
-            if [ -f "$i/lib$j.so" ]; then
-                addToRPath $i
+    for dir in "${libDirs[@]}"; do
+        for file in "$dir"/lib*.so; do
+            if [ "${libs[${file##*/}]}" ]; then
```

Comment:
`file` must end with `.so` so it can not end with `/`, even if `lib*.so` is a directory, which is not necessary to support, and testing if an entry is a file or a directory is slow.

`${file##*/}` is needed to take the basename of the file because `libs` contains only file names and `dir/*` returns `dir/a`, `dir/b`, etc.; calling `basename $file` here is too slow since `basename` is not a bash builtin.